include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,295 @@
+      switching_behaviors_ = true;  // indicate transition is in progress
+    } else
+    {
+      LOG_WARN("Invalid behavior selected: %d", next_idx);
+    }
+  }
+
+  /**
+   * @brief Constructs and initializes a behavior
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   * @return unique pointer to the behavior if initialization was successful,
+   *         nullptr otherwise
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  std::unique_ptr<BehaviorType> InitBehavior(ConstructorArgs &&... args)
+  {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = std::make_unique<BehaviorType>(args...);
+    if (b->Init())
+    {
+      b->set_initialized();
+      return std::move(b);
+    } else { return nullptr; }
+  }
+
+  /**
+   * @brief Add a behavior to the behaviors list and initialize it
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  void AddBehavior(ConstructorArgs &&... args)
+  {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = InitBehavior<BehaviorType>(args...);
+    if (b)
+    {
+      names_.push_back(b->get_name());
+      behaviors_.emplace_back(std::move(b));
+    } else
+    {
+      LOG_WARN(
+          "%s behavior could not be initialized and will not be added to behaviors list.",
+          b->get_name().c_str());
+    }
+  }
+
+  /**
+   * @brief Replaces the default behavior with a user-defined one
+   * @note This is included to provide support for custom default robot
+   *       behaviors. If this function is not called, the default behavior will
+   *       be a `kodlab::OffBehavior`.
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  void SetDefaultBehavior(ConstructorArgs &&... args)
+  {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = InitBehavior<BehaviorType>(args...);
+    if (b)
+    {
+      names_.front() = b->get_name();
+      behaviors_.front() = std::move(b);
+    } else
+    {
+      LOG_WARN(
+          "%s behavior could not be initialized and will not be set as default behavior.",
+          b->get_name().c_str());
+    }
+  }
+
+  /**
+   * @brief Reset the behaviors list back to default
+   */
+  void ResetBehaviors()
```

Comment:
`EraseBehaviors` Reset makes me think the it just sets the behaviors back to their default state rather than removing all of them.

Reply:
Renamed in d5a6010ee947e62b6bbd122659ed3fcc02ff3224